azurerm/internal/services/network/express_route_connection_resource.go

```diff
@@ -0,0 +1,349 @@
+	azSchema "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceExpressRouteConnection() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceExpressRouteConnectionCreateUpdate,
+		Read:   resourceExpressRouteConnectionRead,
+		Update: resourceExpressRouteConnectionCreateUpdate,
+		Delete: resourceExpressRouteConnectionDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Importer: azSchema.ValidateResourceIDPriorToImport(func(id string) error {
+			_, err := parse.ExpressRouteConnectionID(id)
+			return err
+		}),
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.ExpressRouteConnectionName,
+			},
+
+			"express_route_circuit_peering_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.ExpressRouteCircuitPeeringID,
+			},
+
+			"express_route_gateway_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.ExpressRouteGatewayID,
+			},
+
+			"authorization_key": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.IsUUID,
+			},
+
+			"enable_internet_security": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+
+			// Note: when the `routing` property isn't specified, the `associated_route_table_id` property and the `propagated_route_table` property would be set with the default value.
+			// As the `routing` property has default value, so it has to add the `ConfigMode` attribute to roll back to the default value after the `routing` property is specified.
+			"routing": {
+				Type:       schema.TypeList,
+				Optional:   true,
+				Computed:   true,
+				MaxItems:   1,
+				ConfigMode: schema.SchemaConfigModeAttr,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						// Note: When the `associated_route_table_id` property isn't specified, it has default value. So it has to add the `Computed` attribute to ignore the diff.
+						// But if the `Computed` attribute is added, it cannot be rolled back to default value after this property is specified. So the `Computed` attribute has to be removed from schema.
+						"associated_route_table_id": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validation.Any(validate.HubRouteTableID, validation.StringIsEmpty),
```

Comment:
I personally prefer adding the `Computed` back here.

Reply:
So you mean we have to abandon the ability to set the default value back?